api/proto/metrics/metrics.proto

```diff
@@ -177,12 +177,39 @@ message appCpuMetric {
   uint64 total_ns = 7;                 // like 'total' but in nanoseconds. Overflow may happen after ~40 years without EVE restart
 }
 
-// StorageMetric   At the moment not used
+// StorageDeviceMetrics - contains device statistics from ZFS storage
+message StorageDeviceMetrics {
+  uint64 alloc = 1;             // space allocated (in byte)
+  uint64 total = 2;             // total capacity (in byte)
+  uint64 deflated_space = 3;    // deflated capacity (in byte)
+  uint64 replaceable_size = 4;  // replaceable dev size (in byte)
+  uint64 expandable_size = 5;   // expandable dev size (in byte)
+  uint64 read_errors = 6;       // counter read count errors
+  uint64 write_errors = 7;      // counter write errors
+  uint64 checksum_errors = 8;  // counter checksum errors
+  uint64 bytes_read = 9;       // bytes read (in byte)
+  uint64 bytes_write = 10;      // bytes written (in byte)
+  uint64 ops_count_read = 11;   // Number of read ops
+  uint64 ops_count_write = 12;  // Number of write ops
+  uint64 ios_in_progress = 13;  // number of I/Os currently in progress.
+  uint64 read_ticks = 14;       // The total number of milliseconds spent by all reads.
+  uint64 write_ticks = 15;      // The total number of milliseconds spent by all writes.
+  uint64 ios_total_ticks = 16;  // The number of milliseconds spent doing I/Os.
+  // weighted_io_ticks is the weighted number of milliseconds spent doing I/Os.
+  // This can also be used to estimate average queue wait time for requests.
+  uint64 weighted_io_ticks = 17;
+}
+
+// StorageMetric - contains zpool statistics including child devices
 message StorageMetric {
-  string pool_name = 1;
-  // Depreated tag 2
-  // Depreated tag 3
-  // Depreated tag 4
+  // Deprecated = 1;
+  // Deprecated = 2;
+  // Deprecated = 3;
+  // Deprecated = 4;
+  string display_name = 5;                       // device/dataset/pool name in storage
```

Comment:
Don't we want the same identifying information as we have for StorageInfo which is a pool_name and no display_name?

Reply:
The name of the zpool is enough to identify it, as it is unique within zfs. Also, there are no other aliases for a particular zpool in EVE ("persist" is "persist" everywhere).

I agree that perhaps naming a variable as "display_name" is not a super good solution, but since this variable can store different names for different types(zpool, raid, vdev), I thought that would be acceptable.